### PR TITLE
Use client.TransportFor to set an etcd transport

### PR DESCRIPTION
**Pull request by smarterclayton**
Increase max idle connections as per Kubernetes #7353, and
stop using Etcd's slightly crazy initialization.

Also define a default dialer and set max idle connections for
internal client commands.

@liggitt

### CODE REVIEW

**Comment by smarterclayton** [comment]
[test]

File: pkg/cmd/server/api/helpers.go [hunk 1]
```diff
@@ -170,6 +175,20 @@ func GetOpenShiftClient(kubeConfigFile string) (*client.Client, *kclient.Config,
...
+// for use by infrastructure components.
+func DefaultClientTransport(rt http.RoundTripper) http.RoundTripper {
+	transport := rt.(*http.Transport)
+	dialer := &net.Dialer{
+		Timeout:   30 * time.Second,
+		KeepAlive: 30 * time.Second,
+	}
+	transport.Dial = dialer.Dial
+	// Hold open more internal idle connections
+	transport.MaxIdleConnsPerHost = 50
```
Comment (liggitt) [inline pkg/cmd/server/api/helpers.go]: this is per remote-host?

File: pkg/cmd/server/etcd/etcd.go [hunk 2]
```diff
@@ -71,39 +74,48 @@ func RunEtcd(etcdServerConfig *configapi.EtcdConfig) {
...
+	transport := rt.(*http.Transport)
+	dialer := &net.Dialer{
+		Timeout: 30 * time.Second,
+		// Lower the keep alive for connections.
+		KeepAlive: 1 * time.Second,
+	}
+	transport.Dial = dialer.Dial
+	// Because watches are very bursty, defends against long delays
+	// in watch reconnections.
+	transport.MaxIdleConnsPerHost = 500
```
Comment (liggitt) [inline pkg/cmd/server/etcd/etcd.go]: does this mean the API server only supports 500 open watches?

**Comment by liggitt** [comment]
a couple questions about the numbers picked, but the client building LGTM

File: pkg/cmd/server/etcd/etcd.go (hunk 2 above)
Comment (smarterclayton) [inline pkg/cmd/server/etcd/etcd.go]: No, it means it only holds on to 500 connections after they are not used

File: pkg/cmd/server/api/helpers.go (hunk 1 above)
Comment (smarterclayton) [inline pkg/cmd/server/api/helpers.go]: Yes

Comment (smarterclayton) [inline pkg/cmd/server/api/helpers.go]: We scale with number of nodes here basically.

**Comment by smarterclayton** [comment]
[merge]

**Pull request merged**